packages/in_app_purchase/in_app_purchase_android/android/src/test/java/io/flutter/plugins/inapppurchase/InAppPurchasePluginTest.java

```diff
@@ -37,4 +52,28 @@ public void registerWith_doNotCrashWhenRegisterContextIsActivity_V1Embedding() {
     when(activity.getApplicationContext()).thenReturn(mockApplication);
     InAppPurchasePlugin.registerWith(mockRegistrar);
   }
+
+  // The PROXY_PACKAGE_KEY value of this test (io.flutter.plugins.inapppurchase) should never be changed.
+  // In case there's a strong reason to change it, please inform the current code owner of the plugin.
+  @Test
+  public void registerWith_proxyIsSet_V1Embedding() {
+    when(mockRegistrar.context()).thenReturn(activity);
+    when(activity.getApplicationContext()).thenReturn(mockApplication);
+    InAppPurchasePlugin.registerWith(mockRegistrar);
+    Mockito.verify(mockIntent).putExtra(PROXY_PACKAGE_KEY, "io.flutter.plugins.inapppurchase");
+    assertEquals("io.flutter.plugins.inapppurchase", BuildConfig.APPLICATION_ID);
```

Comment:
nit: what do you think about bumping the Gradle version of the plugin, so it can use `LIBRARY_PACKAGE_NAME`?  Also, a comment will be appreciated for the new person:


"BuildConfig" cannot be used without reflection because whether to read `BuildConfig.APPLICATION_ID` or `LIBRARY_PACKAGE_NAME` depends on the "APP's" Android Gradle plugin version.  Newer versions of AGP use `LIBRARY_PACKAGE_NAME`, whereas older ones use `BuildConfig.APPLICATION_ID`.